CONTRIBUTING.md

```diff
@@ -1,65 +1,231 @@
+   [Google Developer Documentation Style Guide][style-guide].
+   Focus on the [**Voice and tone**][gs-tone]
+   and the [**Highlights**][gs-highlights]
+   pages. They provide the best starting points.
+1. Using `git`, [commit changes](https://git-scm.com/docs/git-commit) often.
+   It simplifies reverting a specific change.
+   Reviewers can choose to review changes made since their last review and
+   shorten review cycles.
+1. When your draft is complete,
+   [push the change to GitHub][push-change].
+
+### Submit a PR and request a review
+
+1. [Create a pull request][create-pr]
+   and verify that you have the right target branch.
+   Most PRs are merged into the `main` branch.
+   If you need to target an upcoming release or new feature,
+   your PR might be merged a dedicated branch with a different name.
+
+1. Write the PR summary using the following guidelines:
+
+   1. Describe the change.
+      Include a brief description of what's changing in the pull request.
+      Do this even if you link to a relevant issue.
+
+   1. Link to any related issues. If the PR should fix a particular issue,
+      include a line that states `Fixes #<issue number>`.
+      (Including this phrase links the PR to the issue.
+      When you close the PR, GitHub closes the issue).
+      If your PR fixes multiple issues,
+      include this phrase in whole for each issue.
+
+   1. Point to specific changes. If your PR is large,
+      consider pointing the reviewer to the specific files or lines
+      you'd like the reviewer to check.
+
+1. If your PR isn't ready for review,
+   [set it to Draft][set-draft] and
+   [apply the **st.WIP** label][apply-wip-label].
+
+1. Click **Create pull request**. This starts the review process.
+
+### Respond to reviewer feedback
+
+1. Read and address comments from the reviewer.
+   Acknowledge all comments with at least add a [reaction emoji][reaction-emoji].
+   If the changes are technically or grammatically correct, make them.
+   Reply to comments that you want to discuss.
+   Remember that all comments are public. Keep your replies civil and relevant.
+
+1. When your PR is ready for another round of feedback,
+   re-request a review or write a comment with `PTAL @[reviewer]`.
+   (PTAL means _please take another look_.)
+
+1. Once your PR is approved, a reviewer [merges it][merges-pr].
+
+## Reviewers
+
+### General guidelines for reviews
+
+* Technical review comes before content review, or "copy", review.
+* Lean toward velocity. Aim for _improvement_ over absolute _perfection_.
+* Check the whole page. Sometimes new content contradicts unchanged
+  content elsewhere on the same page.
+* Prefer giving an overall note when there's a repeated problem,
+  rather than commenting on every instance.
+* Try to **approve** with suggestions wherever possible.
+  Use **request changes** only when the change can harm an end user's work or
+  violates [Flutter's Code of Conduct][code-of-conduct].
```

Comment:
I would expect that if a PR violates the code of conduct, that we would just close it _and_ close comments on it.